iceoryx2-cal/src/zero_copy_connection/common.rs

```diff
@@ -765,13 +787,30 @@ pub mod details {
                 segment_details.used_chunk_list.remove_all(|index| {
                     callback(PointerOffset::from_offset_and_segment_id(
                         index * segment_details.sample_size.load(Ordering::Relaxed),
-                        SegmentId::new(n as u8),
+                        self.segment_id_from_index(n),
                     ))
                 });
             }
         }
     }
 
+    impl<Storage: DynamicStorage<SharedManagementData>> Sender<Storage> {
+        fn segment_id_from_index(&self, index: usize) -> SegmentId {
+            let storage = self.storage.get();
+            let number_of_segments = storage.number_of_segments as usize;
+            // the segment details contain an entry for every channel in every segment.
+            // so it is in memory a vector of vector where the first index is the
+            // segment id and the second is channel id. this vector is, to make the
```

Comment:
The first index is the channel id and the second index is the segment id, right?